pkg/apiserver/resthandler.go

```diff
@@ -147,7 +147,20 @@ func getRequestOptions(req *restful.Request, scope RequestScope, kind string, su
 		newQuery[subpathKey] = []string{req.PathParameter("path")}
 		query = newQuery
 	}
-	return queryToObject(query, scope, kind)
+	versioned, err := scope.Creater.New(scope.ServerAPIVersion, kind)
+	if err != nil {
+		// programmer error
+		return nil, err
+	}
+	if err := scope.Codec.DecodeParametersInto(query, versioned); err != nil {
```

Comment:
IMO, it is correct to do the deserialization and conversion separately. This is because the conversion is to a format that the *infrastructure* needs; therefore that logic should be locate with the infrastructure. We should continue to mutate things until the *versioned* object is passed into the REST.List() method (if it's not already). If we do deserialization and conversion at the same time, then we'll have to do *two* deserializations, one for the infrastructure code and one for the strategy code.